library/aes.c

```diff
@@ -1015,6 +1036,7 @@ int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
     AES_VALIDATE_RET( output != NULL );
     AES_VALIDATE_RET( mode == MBEDTLS_AES_ENCRYPT ||
                       mode == MBEDTLS_AES_DECRYPT );
+    (void) mode;
```

Comment:
Is this cast really needed?

Reply:
Yes it is. `mode` is not used when MBEDTLS_AES_ONLY_ENCRYPT is enabled and it will cause compiler warning.

Reply:
True, I was thinking about the line before this, but it seems that when `MBEDTLS_CHECK_PARAMS` isn't defined, the `AES_VALIDATE_RET` will be plain `do { } while( 0 )` and omits the usage of `mode`